### DISCUSSION TITLE
Dropping support for NVENC on NVIDIA Kepler-based GPUs

### DISCUSSION THREAD

**Discussion by RytoEX** [obsproject/obs-studio#11058] [Announcements]
# Overview

OBS Studio 30.2 will be the last release to support NVENC on NVIDIA Kepler-based (GTX 600, 700, some 800, and Quadro K-series) GPUs. OBS Studio 31 will not support NVENC on NVIDIA Kepler GPUs.

# Why Drop Kepler GPUs?

We originally considered dropping support for Kepler with OBS Studio 28.1 in October 2022 when we updated the NVENC SDK version to add AV1 support. In the end, we decided that it wasn't right to drop hardware support in a minor release, and we would revisit the idea in the next major release, OBS Studio 29. When OBS Studio 29 was gearing up, we simply forgot to revisit the subject, and then forgot again with OBS Studio 30.

We actually maintained NVENC support for Kepler GPUs by way of [a rather hacky bit of code](https://github.com/obsproject/obs-studio/commit/53f4627b0c8a08f6b430b77c8400b289ee0b36ed). However, that compatibility hack has made maintaining and debugging the rest of the NVENC code a bit nightmarish, and it has made introducing new features difficult. There have been multiple times when we have run into issues in part due to trying to maintain this compatibility. We have concluded that the maintenance burden of keeping this compatibility is too high to continue justifying it.

NVIDIA ended active support of Kepler GPUs years ago.

# When Will Things Happen?

## Timetable

| Timeframe          | Action                                                                             |
| ------------------ | ---------------------------------------------------------------------------------- |
| Soon               | Removal of the compatibility hack and Kepler support for NVENC                     |
| Later              | Stable release of next major OBS version                                           |

## Soon

We will remove the code that enables NVENC on Kepler GPUs.

## Later

At a future point in time, we will release the next major version of OBS that will not support NVENC on Kepler GPUs.
See:
* https://github.com/obsproject/obs-studio/pull/10536

# How Does This Affect Me?

## Users

Users of the first-party OBS distributions on Windows, Flatpak, or Ubuntu 24.04 with Kepler GPUs will no longer be able to use NVENC to record or stream. They will still be able to run OBS and use all other functions. If you wish to continue using NVENC on Kepler GPUs, OBS Studio 30.2 or older will still be available and still work.

**Comment by DealsBeam** [comment]
Can we please get support for Intel Deep Link technology https://www.intel.com/content/www/us/en/architecture-and-technology/adaptix/deep-link.html

**Comment by Lee-7723** [comment]
bro really said: "i forgor 💀" 🤣🤣🤣

**Comment by RytoEX** [comment]
This change has shipped with OBS Studio 31.0.0.
